synapse/media/oembed.py

```diff
@@ -98,7 +98,7 @@ def get_oembed_url(self, url: str) -> Optional[str]:
         # No match.
         return None
 
-    def autodiscover_from_html(self, tree: "etree.Element") -> Optional[str]:
+    def autodiscover_from_html(self, tree: "etree._Element") -> Optional[str]:
```

Comment:
What's the difference between `Element` and `_Element`? Perhaps the annotation should have always been `_Element`, and `_Element` isn't publicly exported?

Reply:
`Element` is a function which creates an `_Element`:

```pycon
>>> from lxml import etree
>>> etree.Element
<cyfunction Element at 0x10d478c70>
>>> etree._Element
<class 'lxml.etree._Element'>
```

Reply:
Aha!